service/history/outbound_queue_factory.go

```diff
@@ -135,17 +134,18 @@ func NewOutboundQueueFactory(params outboundQueueFactoryParams) QueueFactory {
 					key.Destination,
 				),
 				SettingsFn: func() dynamicconfig.CircuitBreakerSettings {
-					nsName, err := params.NamespaceRegistry.GetNamespaceName(namespace.ID(key.NamespaceID))
-					if err != nil {
-						// This is intentionally not failing the function in case of error. The circuit
-						// breaker is agnostic to Task implementation, and thus the settings function is
-						// not expected to return an error. Also, in this case, if the namespace registry
-						// fails to get the name, then the task itself will fail when it is processed and
-						// tries to get the namespace name.
-						readNamespaceErrors.With(metricsHandler).
-							Record(1, metrics.ReasonTag(metrics.ReasonString(err.Error())))
-					}
-					return circuitBreakerSettings(nsName.String(), key.Destination)
+					// This is intentionally not failing the function in case of error. The circuit
+					// breaker is agnostic to Task implementation, and thus the settings function is
+					// not expected to return an error. Also, in this case, if the namespace registry
+					// fails to get the name, then the task itself will fail when it is processed and
+					// tries to get the namespace name.
```

Comment:
nit: you could probably move the comment to the function but this okay too.

Reply:
I think here's better as it is in the context of the caller since it will be called in a few different places.